saleor/graphql/webhook/subscription_types.py

```diff
@@ -1161,12 +1162,33 @@ def resolve_order(root, info: ResolveInfo):
 
 
 class FulfillmentCreated(SubscriptionObjectType, FulfillmentBase):
+    notify_customer = graphene.Boolean(
+        description="If true, send a notification to the customer." + ADDED_IN_316,
+        required=True,
+    )
+
     class Meta:
+        doc_category = DOC_CATEGORY_ORDERS
         root_type = "Fulfillment"
-        enable_dry_run = True
+        enable_dry_run = False
```

Comment:
As far as I understand this means, that any fulfillment event that will have `notifyCustomer` will not work with dryRun. 
Instead of removing this part of the functionality, have you considered a change in `webhookDryRun` to keep a dry run for fulfillment? 

We could have a map, that for specific objects, changes the structure of the subscribable object. We could set `notifyCustomer` as a dummy value. 
Something like (`saleor/graphql/webhook/mutations/webhook_dry_run.py`):

```python
    # new method added
    @classmethod
    def get_subscribable_object(cls, db_object, event_type):
        if event_type == "fulfillment_created":
            return {
                "fulfillment": db_object,
                "notify_customer": True,
            }
        return db_object

    @classmethod
    def perform_mutation(cls, _root, info, **data):
        event_type, object, query = cls.validate_input(info, **data)
        payload = None
        if all([event_type, object, query]):
            request = info.context
            # line below added
            subscribable_object = cls.get_subscribable_object(object, event_type)
            # replace usage of object with subscribable_object in line below
            payload = generate_payload_from_subscription(
                event_type, subscribable_object, query, request
            )
        return WebhookDryRun(payload=payload)
```
(Don't suggest with the structure, I don't think that the mapper should be placed in mutation code. I've placed it here only to show how we can solve the problem without turning off the dryRun)

Reply:
I think this is a good point, and there should be a task for that - to make all subscriptions work in dry run